util/common/metrics.go

```diff
@@ -27,14 +38,71 @@ func StartSendingMetrics(receiver string, duration, sendingInterval time.Duratio
 			err = SendCollectDMetrics(metricPerInterval, sendingInterval, duration)
 		case "emf":
 			err = SendEMFMetrics(metricPerInterval, metricLogGroup, metricNamespace, sendingInterval, duration)
+		case "app_signals":
+			err = SendAppSignalMetrics() //does app signals have dimension for metric?
+		case "traces":
+			err = SendAppTraceMetrics(duration) //does app signals have dimension for metric?
+
 		default:
 		}
-
 	}()
 
 	return err
 }
 
+func SendAppTraceMetrics(duration time.Duration) error {
+	baseDir := getBaseDir()
+
+	for i := 0; i < int(duration/(5*time.Second)); i++ {
```

Comment:
For this I think using [TimeTicker](https://pkg.go.dev/time#NewTicker) would be easier to maintain and will be less flakey.  Here is an [example](https://github.com/aws/amazon-cloudwatch-agent-test/blob/d4e556e0a638a1b8f6fd9b1f763535a08322b816/util/common/traces/xray/generator.go#L28-L41) from traces-test

Reply:
I still insist by this